include/solvers/mpm_base.h

```diff
@@ -260,6 +269,12 @@ class MPMBase : public MPM {
   //! Locate particles
   bool locate_particles_{true};
 
+  //! Absorbing Boundary Variables
+  bool absorbing_boundary_{false};
+  std::vector<std::shared_ptr<mpm::AbsorbingConstraint>> absorbing_constraint_;
+  std::vector<int> absorbing_nset_id_;
+  mpm::Position position_{mpm::Position::None};
```

Comment:
We don't need to save position_ right? Just define a local variable will work I guess.